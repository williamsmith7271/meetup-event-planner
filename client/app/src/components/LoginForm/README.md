## LoginForm Component
A component that ...

### Example

```js
<LoginForm />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
