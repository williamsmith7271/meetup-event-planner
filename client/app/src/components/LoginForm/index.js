import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Form from 'grommet-udacity/components/Form';
import Heading from 'grommet-udacity/components/Heading';
import FormField from 'grommet-udacity/components/FormField';
import FormFields from 'grommet-udacity/components/FormFields';
import Footer from 'grommet-udacity/components/Footer';
import Button from 'grommet-udacity/components/Button';
import Box from 'grommet-udacity/components/Box';
import { AuthFormFooter } from 'components';
import calculatedError from './utils/error';

const LoginForm = ({
  passwordInput,
  emailInput,
  onSubmit,
  invalid,
}) => (
  <Box
    className={styles.loginForm}
    pad={{ horizontal: 'large' }}
  >
    <Form>
      <Heading strong align="center">
        Meetup Event Planner
      </Heading>
      <Heading align="center" tag="h5">
        Login
      </Heading>
      <FormFields>
        <FormField
          help="How should we get in touch with you?"
          error={calculatedError(emailInput)}
          label="Email *"
          htmlFor="emailInput"
          className={styles.formField}
        >
          <input
            {...emailInput}
            required
            autoFocus
            id="emailInput"
            name="email"
            type="email"
            className={styles.input}
          />
        </FormField>
        <FormField
          help="Gotta be secure"
          error={calculatedError(passwordInput)}
          label="Password *"
          htmlFor="passwordInput"
          className={styles.formField}
        >
          <input
            {...passwordInput}
            required
            name="password"
            id="passwordInput"
            type="password"
            className={styles.input}
          />
        </FormField>
      </FormFields>
      <Footer pad={{ vertical: 'medium' }} align="center">
        <Button onClick={invalid ? null : onSubmit} fill label="Submit" primary />
      </Footer>
      <AuthFormFooter text="Need an Account?" link="/signup" />
    </Form>
  </Box>
);

LoginForm.propTypes = {
  passwordInput: PropTypes.object.isRequired,
  emailInput: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  invalid: PropTypes.bool.isRequired,
};

export default cssModules(LoginForm, styles);
