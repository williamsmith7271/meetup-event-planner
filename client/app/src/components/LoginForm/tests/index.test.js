import LoginForm from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { fields } from './mocks';

describe('<LoginForm />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <LoginForm
        {...fields}
        onSubmit={e => e}
        invalid
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
