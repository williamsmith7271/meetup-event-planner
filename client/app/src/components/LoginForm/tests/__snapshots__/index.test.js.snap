exports[`<LoginForm /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad={
    Object {
      "horizontal": "large"
    }
  }
  responsive={true}
  tag="div">
  <Form
    compact={false}
    fill={false}
    pad="none">
    <Heading
      align="center"
      strong={true}
      tag="h1">
      Meetup Event Planner
    </Heading>
    <Heading
      align="center"
      tag="h5">
      Login
    </Heading>
    <FormFields>
      <FormField
        help="Enter the email you used to create your account"
        htmlFor="emailInput"
        label="Email *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autoFocus={true}
          autofilled={false}
          dirty={true}
          id="emailInput"
          initialValue=""
          invalid={false}
          name="email"
          pristine={false}
          required={true}
          touched={true}
          type="email"
          valid={true}
          value="demo@udacity.com"
          visited={true} />
      </FormField>
      <FormField
        help="Enter the password you used to create your account"
        htmlFor="passwordInput"
        label="Password *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={true}
          id="passwordInput"
          initialValue=""
          invalid={false}
          name="password"
          pristine={false}
          required={true}
          touched={false}
          type="password"
          valid={true}
          value="Password123!"
          visited={false} />
      </FormField>
    </FormFields>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium"
        }
      }
      responsive={false}>
      <Button
        fill={true}
        label="Submit"
        onClick={null}
        primary={true}
        type="button" />
    </Footer>
    <undefined
      link="/signup"
      text="Need an Account?" />
  </Form>
</Box>
`;
